### ISSUE TITLE
Karma.js/ Jasmine unit testing error: "templates" is not available

### ISSUE THREAD

**Issue by davidhlee** [pitr/angular-rails-templates#30]
Hi I'm having an issue using angular-rails-templates with Karma.js integration tests. Angular-rails-templates requires that you include "templates" as a dependency in the main app module:

``` javascript
angular.module('MyApp', ['ngRoute', 'templates'...]
```

Karma requires that you load all angular modules used in the app in the karm.conf.js "files" list up front, however there are no angular-rails-templates js files to link to. As a result I can't load my main app module into my Karma tests, and I get this error when I try to run them:

```
Error: [$injector:modulerr] Failed to instantiate module MyApp due to:
Error: [$injector:modulerr] Failed to instantiate module templates due to:
Error: [$injector:nomod] Module 'templates' is not available!
```

Here's where the error is thrown:

``` javascript
describe('my_controller', function(){
  beforeEach(function() {
    module('MyApp'); // Error occurs here
  }
});
```

Basically, Karma is trying to load the "MyApp" module, which depends on the "templates" dependency, which Karma does not know about. Any ideas as to how I can keep this reference to "templates" in my app's main angular.module dependency list while being able to run Karma tests against it?

Thanks!

**Comment by dmarsupastar** [comment]
@kimardenmiller you might want to check out this post http://codetunes.com/2014/karma-on-rails/. It will tell you how to precompile .slim files

**Comment by kimardenmiller** [comment]
@dmarsupastar , thanks. Very helpful! :smile:

**Comment by gottfrois** [comment]
Oh man... just spend hours on this and you saved my day ! Thx
